config/go.d/prometheus.conf

```diff
@@ -0,0 +1,160 @@
+#
+#
+# [ GLOBAL ]
+# These variables set the defaults for all JOBs, however each JOB may define its own, overriding the defaults.
+#
+# The GLOBAL section format:
+# param1: value1
+# param2: value2
+#
+# Currently supported global parameters:
+#  - update_every
+#    Data collection frequency in seconds. Default: 1.
+#
+#  - autodetection_retry
+#    Re-check interval in seconds. Attempts to start the job are made once every interval.
+#    Zero means not to schedule re-check. Default: 0.
+#
+#  - priority
+#    Priority is the relative priority of the charts as rendered on the web page,
+#    lower numbers make the charts appear before the ones with higher numbers. Default: 70000.
+#
+#
+# [ JOBS ]
+# JOBS allow you to collect values from multiple sources.
+# Each source will have its own set of charts.
+#
+# IMPORTANT:
+#  - Parameter 'name' is mandatory.
+#  - Jobs with the same name are mutually exclusive. Only one of them will be allowed running at any time.
+#
+# This allows autodetection to try several alternatives and pick the one that works.
+# Any number of jobs is supported.
+#
+# The JOBS section format:
+#
+# jobs:
+#   - name: job1
+#     param1: value1
+#     param2: value2
+#
+#   - name: job2
+#     param1: value1
+#     param2: value2
+#
+#   - name: job2
+#     param1: value1
+#
+#
+# [ List of JOB specific parameters ]:
+#  - url
+#    Server URL.
+#    Syntax:
+#      url: http://localhost:80
+#
+#  - username
+#    Username for basic HTTP authentication.
+#    Syntax:
+#      username: tony
+#
+#  - password
+#    Password for basic HTTP authentication.
+#    Syntax:
+#      password: stark
+#
+#  - proxy_url
+#    Proxy URL.
+#    Syntax:
+#      proxy_url: http://localhost:3128
+#
+#  - proxy_username
+#    Username for proxy basic HTTP authentication.
+#    Syntax:
+#      username: bruce
+#
+#  - proxy_password
+#    Password for proxy basic HTTP authentication.
+#    Syntax:
+#      username: wayne
+#
+#  - timeout
+#    HTTP response timeout.
+#    Syntax:
+#      timeout: 1
+#
+#  - method
+#    HTTP request method.
+#    Syntax:
+#      method: GET
+#
+#  - body
+#    HTTP request method.
+#    Syntax:
+#      body: '{fake: data}'
+#
+#  - headers
+#    HTTP request headers.
+#    Syntax:
+#      headers:
+#        X-API-Key: key
+#
+#  - not_follow_redirects
+#    Whether to not follow redirects from the server.
+#    Syntax:
+#      not_follow_redirects: yes/no
+#
+#  - tls_skip_verify
+#    Whether to skip verifying server's certificate chain and hostname.
+#    Syntax:
+#      tls_skip_verify: yes/no
+#
+#  - tls_ca
+#    Certificate authority that client use when verifying server certificates.
+#    Syntax:
+#      tls_ca: path/to/ca.pem
+#
+#  - tls_cert
+#    Client tls certificate.
+#    Syntax:
+#      tls_cert: path/to/cert.pem
+#
+#  - tls_key
+#    Client tls key.
+#    Syntax:
+#      tls_key: path/to/key.pem
+#
+#
+# [ JOB defaults ]:
+#  timeout: 5
+#  method: GET
+#  not_follow_redirects: no
```

Comment:
Don't like this key. Suggestion: `follow_redirects: yes`

Reply:
well, i need something that defaults to _follow_ if not set 😄 

I just embed web struct

https://github.com/netdata/go.d.plugin/blob/359f7334899f691a26701d57b0075471b8b98a80/modules/wmi/wmi.go#L44-L53

And it has all web related options with ok defaults

https://github.com/netdata/go.d.plugin/blob/359f7334899f691a26701d57b0075471b8b98a80/pkg/web/request.go#L12-L22

https://github.com/netdata/go.d.plugin/blob/359f7334899f691a26701d57b0075471b8b98a80/pkg/web/client.go#L15-L20

Reply:
THanks!